src/lib.rs

```diff
@@ -221,6 +235,51 @@ macro_rules! get_function {
 get_function!(name: get_mut, mutability: mut);
 get_function!(name: get, mutability: );
 
+/// Macro to parametrise over mutability in get_node methods.
+macro_rules! get_node_function {
+    (
+        name: $name:ident,
+        mutability: $($mut_:tt)*
+    ) => {
+        id!(fn $name<'a, K, V>(
+            trie: &'a $($mut_)* Trie<K, V>,
+            mut key_fragments: NibbleVec
+        ) -> Option<&'a $($mut_)* Trie<K, V> > where K: TrieKey {
+            // Handle retrieval at the root.
+            if key_fragments.len() == 0 {
+                println!("hello");
+                return Some(trie);
+            }
+
+            let bucket = key_fragments.get(0) as usize;
+
+            match trie.children[bucket] {
+                None => None,
+                Some(ref $($mut_)* existing_child) => {
+                    match match_keys(&key_fragments, &existing_child.key) {
+                        KeyMatch::Full => Some(existing_child),
+                        KeyMatch::Partial(idx) => {
+                            let new_key_fragments = key_fragments.split(idx);
+                            $name(& $($mut_)* *existing_child, new_key_fragments)
+                        },
+                        KeyMatch::FirstPrefix => Some(existing_child),
```

Comment:
It sounds like you may be able to use `get_ancestor_value` (its new name) for your use case?

For example, if you fed in `"首次将有女性成为美元纸币的正面人物"`, it would give you the longest prefix that matches a word in the trie.